metagpt/roles/sk_agent.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+@Time    : 2023/9/13 12:23
+@Author  : femto Zheng
+@File    : sk_agent.py
+"""
+from semantic_kernel.planning import SequentialPlanner
+from semantic_kernel.planning.action_planner.action_planner import ActionPlanner
+from semantic_kernel.planning.basic_planner import BasicPlanner
+
+from metagpt.actions import BossRequirement
+from metagpt.actions.execute_task import ExecuteTask
+from metagpt.logs import logger
+from metagpt.roles import Role
+from metagpt.schema import Message
+from metagpt.utils.make_sk_kernel import make_sk_kernel
+
+
+class SkAgent(Role):
+    """
+    Represents an SkAgent implemented using semantic kernel
+
+    Attributes:
+        name (str): Name of the SkAgent.
+        profile (str): Role profile, default is 'sk_agent'.
+        goal (str): Goal of the SkAgent.
+        constraints (str): Constraints for the SkAgent.
+    """
+
+    def __init__(
+        self,
+        name: str = "Sunshine",
+        profile: str = "sk_agent",
+        goal: str = "Execute task based on passed in task description",
+        constraints: str = "",
+        planner_cls=BasicPlanner,
+    ) -> None:
+        """Initializes the Engineer role with given attributes."""
+        super().__init__(name, profile, goal, constraints)
+        self._init_actions([ExecuteTask()])
+        self._watch([BossRequirement])
+        self.kernel = make_sk_kernel()
+
+        # how funny the interface is inconsistent
+        if planner_cls == BasicPlanner:
+            self.planner = planner_cls()
+        elif planner_cls in [SequentialPlanner, ActionPlanner]:
+            self.planner = planner_cls(self.kernel)
```

Comment:
add else raise(unsupported planner)

Reply:
added, but actually I'm thinking to remove the second judgement clause,
make it like this:
```
        if planner_cls == BasicPlanner:
            self.planner = planner_cls()
        else:
            self.planner = planner_cls(self.kernel)
```
because 1. in the future, semantic kernel will add other planner
2. we will add other planner, 
so as long as the planner's constructor signature is planner_cls(self.kernel),
then it should be fine